### PR TITLE
[docs] Add changes to gcc-tips

### PR DESCRIPTION
**Pull request by akshitadixit**
* Removed all images except gcc-cpmparison-2.png
* Added alt text to it

### CODE REVIEW

File: docs/gcc-Tips.rst
Comment (jameslamb) [inline docs/gcc-Tips.rst:35]:
```suggestion
   :alt: picture with a chart grouped by compiler set of options using O2 mtune=core2 as the baseline. All the other 3 options are faster, with O3 M tune equals native being the fastest.
```

I think it might help screen readers to break this phrase up, what do you think?

File: docs/gcc-Tips.rst
Comment (jameslamb) [inline docs/gcc-Tips.rst:30]:
```suggestion
The image below compares the runtime for training with different compiler options to a baseline using LightGBM compiled with `-O2 --mtune=core2`. All three options are faster than that baseline. The best performance was achieved with `-O3 --mtune=native`.
```

Could you expand this description? For figures like this, I like to have explanatory text that tells people what conclusion they're supposed to draw.

**Review by jameslamb — CHANGES_REQUESTED** [review]
Thanks for doing this! Please see my comments below.

File: docs/gcc-Tips.rst
Comment (akshitadixit) [reply docs/gcc-Tips.rst:35]: I checked with a screen reader and only separating the 'M' works well, the equals to operator is being read aloud. Your suggestion is good to go if both occurences of mtune be changed to M tune.

Comment (jameslamb) [reply docs/gcc-Tips.rst:35]: ah right, thank you! I just edited the suggestion to work for both uses of `mtune`.

**Comment by jameslamb** [comment]
The two CI failures were unrelated to this PR, and I just restarted both of them manually. So I think this can be merged once https://github.com/microsoft/LightGBM/pull/4187#discussion_r614971908 is resolved.

**Comment by akshitadixit** [comment]
I just wish I can submit a clean PR for once. Thank you for the patience. I learnt why every senior dev always told me to create new branches and avoid working on master directly.

**Review by jameslamb — APPROVED** [review]
This looks good to me, thanks very much!

@StrikerRUS in case you didn't see on the previous PR, the deletions in this PR were done with @Laurae2 's permission: https://github.com/microsoft/LightGBM/pull/4175#issuecomment-820748957

> I just wish I can submit a clean PR for once.

Don't even worry about it! If you look back through mine or other maintainers' pull requests here, you'll see that even we (people who have been working on this project for years) very rarely have a PR that is merged without any requested changes. It's just a part of the process.

**Comment by StrikerRUS** [comment]
@akshitadixit Please remove actually files from the following folder references for which were removed in this PR.
https://github.com/microsoft/LightGBM/tree/master/docs/_static/images

**Review by StrikerRUS — APPROVED** [review]
LGTM! @akshitadixit We really appreciate your great help!

**Pull request merged**
